### PR TITLE
Let use external nlohmann/json.hpp (ROOT-10993)

### PR DESCRIPTION
**Pull request by linev**
Move existing nlohmann/json.hpp include to builtin location
and use it by default. When `builtin_nlohmannjson` is OFF,
cmake tries to find it in default location.

### CODE REVIEW

**Comment by linev** [comment]
@phsft-bot build with flags -Dbuiltin_nlohmannjson=ON

File: cmake/modules/RootBuildOptions.cmake [hunk 1]
```diff
@@ -93,6 +93,7 @@ ROOT_BUILD_OPTION(builtin_gsl OFF "Build GSL internally (requires network)")
 ROOT_BUILD_OPTION(builtin_llvm ON "Build bundled copy of LLVM")
 ROOT_BUILD_OPTION(builtin_lz4 OFF "Build bundled copy of lz4")
 ROOT_BUILD_OPTION(builtin_lzma OFF "Build bundled copy of lzma")
+ROOT_BUILD_OPTION(builtin_nlohmannjson ON "Use nlohmann/json.hpp file destributed with ROOT")
```
Comment (oshadura) [inline cmake/modules/RootBuildOptions.cmake:96]: I think it should be 'distributed'

**Review by oshadura — COMMENTED** [review]
We tested PR with lcgdev3/dev4 and it is properly discovering external `nlohmann/json.h`. I think we need to enable it for Jenkins CI as a next step (@Axel-Naumann )?

**Comment by linev** [comment]
@phsft-bot build with flags -Dbuiltin_nlohmannjson=ON

**Comment by linev** [comment]
@phsft-bot build with flags -Dbuiltin_nlohmannjson=ON

**Review by oshadura — APPROVED** [review]
I think we can try to go with this approach since it is critical for LHCB, and maybe then I can try to converge to manage external and internal nlohman json parser in the same way?

**Comment by linev** [comment]
@oshadura @Axel-Naumann 
Before I can merge it in master, one should change `Jenkins` configuration, otherwise it will try to disable `builtin_nlohmannjson` for every platform :(

**Comment by linev** [comment]
@phsft-bot build

**Comment by linev** [comment]
Seems to be, all platforms compiled, only for `mac1014` jenkins did not found build node.

**Pull request merged**